source/entities/Creature.h

```diff
@@ -423,21 +428,18 @@ class Creature: public MovableGameEntity
     std::deque<CreatureAction>      mActionQueue;
     std::vector<Tile*>              mVisualDebugEntityTiles;
 
-
-    //! \brief Allows to keep track of the entity we are currently attacking
-    Tile*                           mAttackedTile;
-    GameEntity*                     mAttackedObject;
+    //! \brief Contains the actions that have already been tested to avoid trying several times same action
```

Comment:
yep, that's it. The logic in the idle function is basically to roll a dice and, if the value allows, push an action to test it. IMHO, that's a smart way to handle that but it can result in testing several times the same action (depending on chance). That's to avoid this case that I've done that.